factory/usr/lib/the-modeenv

```diff
@@ -5,10 +5,19 @@ cp /sysroot/run/image.fstab /run/
 # TODO:UC20: move this logic into snap-bootstrap to be incorporated with 
 # initramfs-mounts
 
+if grep -q snapd_recovery_mode=run /proc/cmdline; then
```

Comment:
should we not use `/run/mnt/ubuntu-data/system-data/var/lib/snapd/modeenv` is a primary source of truth?